docs/datamodel/computeds.rst

```diff
@@ -0,0 +1,123 @@
+.. _ref_datamodel_computed:
+
+=========
+Computeds
+=========
+
+.. important::
+
+  This section assumes a basic understanding of EdgeQL. If you aren't familiar
+  with it, feel free to skip this page for now.
+
+Object types can contain *computed* links and properties. Computed properties
+and links are not persisted in the database. Instead, they are evaluated *on
+the fly* whenever that field is queried.
+
+.. code-block:: sdl
+
+  type Person {
+    property name -> str;
+    property all_caps_name := str_upper(__subject__.name);
+  }
+
+Computed fields are are associated with an EdgeQL expression. This expression
+can be an *arbitrary* EdgeQL query. This expression is evaluated whenever the
+field is referenced in a query.
+
+.. _ref_dot_notation:
+
+Dot notation
+------------
+
+The example above used the special keyword ``__subject__`` to refer to
+the current object; it's analogous to ``this`` in many object-oriented
+languages.
+
+However, explicitly using ``__subject__`` is optional here; inside the scope of
+an object type declaration, you can omit it entirely and use the ``.<name>``
+shorthand.
+
+.. code-block:: sdl
+
+  type Person {
+    property first_name -> str;
+    property last_name -> str;
+    property full_name := .first_name ++ ' ' ++ .last_name;
+  }
+
+Type inference
```

Comment:
This section talks more about Cardinality Inference than about Type Inference. So

```suggestion
Type and cardinality inference
```

?